Add merge-trunk-develop-pr action

tomalec opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->
Add merge-trunk-develop-pr action to create a PR to merge `trunk` to `develop` after a release done by `woocommerce/grow/prepare-extension-release`. As automerging with `automerge-released-trunk` may not work with branch protections.


### Screenshots:

![image](https://github.com/woocommerce/grow/assets/17435/56dfeb95-fe43-43cf-8048-c6055b196879)

![image](https://github.com/woocommerce/grow/assets/17435/5cfc7184-c6de-41de-97b2-9f09dc0b023b)



### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Create a test repo that has the ``woocommerce/grow/prepare-extension-release`` set up.
2. Add to trunk another workflow following the README https://github.com/woocommerce/grow/blob/add/merge-trunk-develop-pr/packages/github-actions/actions/merge-trunk-develop-pr/README.md but with the prereleased version `actions-v1.9.1-pre`
```yml
name: PR to merge a released trunk

on:
  pull_request:
    types:
      - closed
    branches:
      - trunk

jobs:
  automerge_trunk:
    runs-on: ubuntu-latest
    steps:
      - uses: woocommerce/grow/merge-trunk-develop-pr@@actions-v1.9.1-pre
```
3. Create a develop branch
4. Create a release with ``woocommerce/grow/prepare-extension-release``. Then merge it - check that a PR to merge `trunk` to `develop` gets created. Like
   - https://github.com/tomalec/grow/pull/27
   - https://github.com/tomalec/grow/actions/runs/5742320916?pr=27
   - https://github.com/tomalec/grow/pull/43
5. Create a release with ``woocommerce/grow/prepare-extension-release``, switch a target branch to something else. Merge the PR. No PR should get created
5. Create a release with ``woocommerce/grow/prepare-extension-release``, close it without merging. No PR should get created
6. Create a PR with `release/1.2.3` as a regular user, then merge it to trunk. No PR should get created


### Additional details:
- Probably after merging this PR, we could tweak the PR checklist to mention using merging the PR created by this action, if applicable

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

For changes related to the `github-actions` package, please use `[actions] changelog: *` labels.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Add merge-trunk-develop-pr action

tomalec committed:
Add merge-trunk-develop-pr action

to create a PR to merge `trunk` to `develop` after a release done by `woocommerce/grow/prepare-extension-release`.
As automerging with `automerge-released-trunk` may not work with branch protections.

eason9487 commented on packages/github-actions/actions/merge-trunk-develop-pr/action.yml:
```diff
@@ -0,0 +1,28 @@
+      # https://api.github.com/users/github-actions%5Bbot%5D
+      run: |
+        git config user.name github-actions[bot]
+        git config user.email 41898282+github-actions[bot]@users.noreply.github.com
```
I'm not pretty sure but maybe these two steps are not necessary because the next step doesn't require any repo files nor operates with `git commit`. Perhaps it's worth trying to see if this action still works without them or at least the second one.

tomalec replied:
Thanks :) I thought GHA needed an explicit username to create PRs.
But Github API calls works out of the box, away from CLI `git` https://github.com/tomalec/grow/actions/runs/5787573264?pr=30 :) 

Removed in https://github.com/woocommerce/grow/pull/71/commits/161b44aefbd32d256ec0fc4d4243f58671655fc9

eason9487 reviewed (approved):
Thanks for the work. LGTM.

This new GH action could be added to [this list](https://github.com/woocommerce/grow/blob/91f51db082f69a8a23409a7c3b7760a16fdfd162/packages/github-actions/README.md#actions-list) as well. (Maybe also add `automerge-released-trunk` action by the way?)

All my comments are not blocking this PR.

tomalec committed:
Add merge actions to the README list

Addresses https://github.com/woocommerce/grow/pull/71#pullrequestreview-1560732454

tomalec committed:
Remove git checkout and config steps

Github API calls does not need local repo.

Addresses https://github.com/woocommerce/grow/pull/71#discussion_r1282970078

tomalec commented:
Thank you for the review, and great suggestions. Applied them all.
